lib/ansible/modules/database/mysql/mysql_user.py

```diff
@@ -493,6 +502,22 @@ def pick(x):
     return output
 
 
+def privileges_get_all(cursor):
+    privileges = []
+    cursor.execute('SELECT DISTINCT PRIVILEGE_TYPE FROM INFORMATION_SCHEMA.USER_PRIVILEGES WHERE IS_GRANTABLE="yes"')
+    grants = cursor.fetchall()
+    for grant in grants:
+        privileges.append(grant[0])
+    return privileges
+
+
+def get_db_version(cursor):
```

Comment:
I think it would be best to refactor `use_old_user_mgmt` to handle all versions manipulation (ie returning a clear string stating which kind of user management we should use, for example `PASSWORD_ONLY`, `PASSWORD_OR_AUTH_STRING`, `MYSQL_8_PLUS` ...)